src/Pages/Goals.cshtml.cs

```diff
@@ -1,13 +1,77 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Security.Claims;
+using ePiggyWeb.DataBase;
+using ePiggyWeb.DataManagement;
+using ePiggyWeb.DataManagement.Entries;
+using ePiggyWeb.DataManagement.Goals;
+using ePiggyWeb.Utilities;
 using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.RazorPages;
 
 namespace ePiggyWeb.Pages
 {
     [Authorize]
     public class GoalsModel : PageModel
     {
+        public IGoalList Goals { get; set; }
+        public decimal Savings { get; set; }
+        private int UserId { get; set; }
+
+        [Required(ErrorMessage = "Required")]
+        [BindProperty]
+        [StringLength(30)]
+        public string Title { get; set; }
+
+        [Required(ErrorMessage = "Required")]
+        [BindProperty]
+        public decimal Amount { get; set; }
+
         public void OnGet()
         {
+            var dataManager = new DataManager();
+            Goals = dataManager.Goals.GoalList;
+            UserId = int.Parse(User.FindFirst(ClaimTypes.Name).Value);
```

Comment:
Am I understanding correctly that you save and get the user id in the cookies so you don't have to access the database to find it?

Reply:
Yep, it is saved in these claims, you've asked about in previous pull request.